pkg/pipeline/CiHandler.go

```diff
@@ -1330,21 +1335,42 @@ func (impl *CiHandlerImpl) FetchCiStatusForTriggerViewForEnvironment(envId int,
 		} else {
 			pipelineId = pipeline.ParentCiPipeline
 		}
-		workflow, err := impl.ciWorkflowRepository.FindLastTriggeredWorkflow(pipelineId)
-		if err != nil && !util.IsErrNoRows(err) {
-			impl.Logger.Errorw("err", "pipelineId", pipelineId, "err", err)
-			return ciWorkflowStatuses, err
+		pipelineIds = append(pipelineIds, pipelineId)
+	}
+	if len(pipelineIds) == 0 {
+		return ciWorkflowStatuses, nil
+	}
+	ciWorkflows, err := impl.ciWorkflowRepository.FindLastTriggeredWorkflowByCiIds(pipelineIds)
```

Comment:
we are fetching results of all CI and not just CI for deploying to relevant environment